### DISCUSSION TITLE
GPIO for water leakage sensor

### DISCUSSION THREAD

**Discussion by Andi99Git** [jomjol/AI-on-the-edge-device#3413] [General]
### The Feature

Is it possible to use a GPIO input as a water leakage sensor?
As soon as the sensor changes the value (open /close), the new status should be sent as a Mqtt message.

**Comment by gitpower2017** [comment]
This is very interesting for me.
I have a TDS sensor that has an analogue output
can this be integrated and if so how?

**Comment by caco3** [comment]
A better approach would be to use a separate esp32 for this.
You can do it even without the need to write software when you use https://github.com/letscontrolit/ESPEasy

**Comment by gitpower2017** [comment]
I am a friend of as few systems as possible.
It is a pity that this is not so easily possible.
I find the combination of water leakage sensor and water meter very suitable

**Reply by caco3** [reply]
The problem is that AFAIK that we are short of pins. There is one free pin which usually gets used for external LEDs. This one could be used.

Looking on the expert settings I can see that GPIO 12 and 13 can be uses as inputs:
![grafik](https://github.com/user-attachments/assets/b60dff7b-773b-40f4-935b-241e9fa39f0c)

How ever I am not sure if they are triggering an MQTT message if they change. Please try it out and let us know.

Further documentation can be found here:
- https://jomjol.github.io/AI-on-the-edge-device-docs/MQTT-API/#gpio
- https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-io12
